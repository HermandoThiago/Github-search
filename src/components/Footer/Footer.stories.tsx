import { Meta, Story } from '@storybook/react';
import Footer, { IPropsFooter } from './Footer';

export default {
    title: 'Components/Footer',
    component: Footer,
    args: {
        children: 'Develop by Hermando Thiago'
    }
} as Meta;

export const Default: Story<IPropsFooter> = () => (
    <Footer>
        Develop by Hermando Thiago
    </Footer>
);

