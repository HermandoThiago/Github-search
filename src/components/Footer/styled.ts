import styled from "styled-components";

export const StyledFooter = styled.footer`
    width: 100%;
    padding: 20px 0;
    position: absolute;
    bottom: 0;
    background-color: #251D3A;
    color: #fff;
    text-align: center;
    font-weight: lighter;
`;
