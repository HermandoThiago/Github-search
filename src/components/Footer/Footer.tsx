import { StyledFooter } from "./styled";

export interface IPropsFooter {
    children: string | React.ReactNode;
}

export default function Footer({ children }: IPropsFooter){
    return (
        <StyledFooter>
            {children}
        </StyledFooter>
    )
}