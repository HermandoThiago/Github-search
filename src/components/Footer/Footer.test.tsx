import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from "./Footer";

describe('Tests Footer', () => {
    it('Render correct text', () => {
        render(
            <Footer>
                Develop by @ ðŸŽ© HermandoThiago
            </Footer>
        )

        screen.getByText(/Develop by @ ðŸŽ© HermandoThiago/i);
    })

    it('should math snapshot', () => {
        const component = renderer.create(
            <Footer>
                Develop by @ ðŸŽ© HermandoThiago
            </Footer>
        ).toJSON();
        expect(component).toMatchSnapshot();
    })
})