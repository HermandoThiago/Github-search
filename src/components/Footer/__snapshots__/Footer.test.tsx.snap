// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Footer should math snapshot 1`] = `
<footer
  className="sc-bczRLJ fZTJYA"
>
  Develop by @ ðŸŽ© HermandoThiago
</footer>
`;
